quesma/quesma/matchers.go

```diff
@@ -36,100 +34,58 @@ func matchedAgainstBulkBody(configuration *config.QuesmaConfiguration, tableReso
-func matchedAgainstPattern(configuration *config.QuesmaConfiguration, sr schema.Registry, indexRegistry table_resolver.TableResolver) mux.RequestMatcher {
-	return mux.RequestMatcherFunc(func(req *mux.Request) bool {
-		indexPattern := elasticsearch.NormalizePattern(req.Params["index"])
-
-		decision := indexRegistry.Resolve(table_resolver.QueryPipeline, indexPattern)
-		table_resolver.TODO("matchedAgainstPattern", decision)
-
-		patterns := strings.Split(req.Params["index"], ",")
-		for i, pattern := range patterns {
-			patterns[i] = elasticsearch.NormalizePattern(pattern)
-		}
-
-		for _, pattern := range patterns {
-			if elasticsearch.IsInternalIndex(pattern) {
-				// We assume that even if one index is an internal Elasticsearch index then the entire query
-				// is an internal Elasticsearch query.
-				logger.Debug().Msgf("index %s is an internal Elasticsearch index, skipping", pattern)
-				return false
-			}
-		}
-
-		if configuration.IndexAutodiscoveryEnabled() {
-			for _, pattern := range patterns {
-				for tableName := range sr.AllSchemas() {
-					if config.MatchName(pattern, string(tableName)) {
-						return true
-					}
-				}
-			}
-		}
-
-		for _, pattern := range patterns {
-			for _, indexConf := range configuration.IndexConfig {
-				if config.MatchName(pattern, indexConf.Name) && configuration.IndexConfig[indexConf.Name].IsClickhouseQueryEnabled() {
-					return true
-				}
-			}
-		}
-
-		return false
-	})
+func matchedAgainstPattern(indexRegistry table_resolver.TableResolver) mux.RequestMatcher {
+	return matchAgainstTableResolver(indexRegistry, table_resolver.QueryPipeline)
 }
 
 // check whether exact index name is enabled
-func matchedExact(cfg *config.QuesmaConfiguration, queryPath bool, indexRegistry table_resolver.TableResolver, pipelineName string) mux.RequestMatcher {
-	return mux.RequestMatcherFunc(func(req *mux.Request) bool {
+func matchAgainstTableResolver(indexRegistry table_resolver.TableResolver, pipelineName string) mux.RequestMatcher {
+	return mux.RequestMatcherFunc(func(req *mux.Request) mux.MatchResult {
 
 		indexName := req.Params["index"]
 
 		decision := indexRegistry.Resolve(pipelineName, indexName)
-		table_resolver.TODO("matchedExact", decision)
-
-		if elasticsearch.IsInternalIndex(req.Params["index"]) {
-			logger.Debug().Msgf("index %s is an internal Elasticsearch index, skipping", req.Params["index"])
-			return false
+		if decision.Err != nil {
+			return mux.MatchResult{Matched: false, Decision: decision}
 		}
-
-		indexConfig, exists := cfg.IndexConfig[req.Params["index"]]
-		if queryPath {
-			return exists && indexConfig.IsClickhouseQueryEnabled()
-		} else {
-			return exists && (indexConfig.IsClickhouseIngestEnabled() || indexConfig.IsIngestDisabled())
+		for _, connector := range decision.UseConnectors {
+			if _, ok := connector.(*table_resolver.ConnectorDecisionClickhouse); ok {
+				return mux.MatchResult{Matched: true, Decision: decision}
+			}
```

Comment:
Similarly to the previous code, this function should return `Matched: true` if ingest to an index is closed - that's the only way Quesma has chance to return an error to the client - now I think it'll just forward it to Elastic instead of blocking.

Reply:
Done